kbase/narrative
# fix kbaseTabs behaving incorrectly when a deletable tab is deleted [PTV-1845]

# Description of PR purpose/changes

The venerable "kbaseTabs" widget is behaving incorrectly in the kbaseExpressionConditionsetBaseWIdget.

Specifically, it is not showing the previous tab when a dynamic tab is closed, as is done when clicking on a cluster.

While fixing the kbaseTabs widget, and expanding the tests to cover additional use case, I discovered at least one additional latent bug (or at least unexpected behavior), a failure to activate the first tab if no tabs are set explicitly as active. 

In addition, there were several features in this already rather complex widget that are not used anywhere in the codebase, and have been removed. They may have been used in the past, but I suspect they were "nice to have" features that were simply added for future utility. However, retaining them would require adding tests to cover them, yet yield no benefit as they are unused.

Finally, tests were expanded to cover common use cases, including the primary issue fixed by this PR.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X

# Testing Instructions

New tests were added to test/unit/spec/widgets/kbaseTabs-spec.js to cover the use case of closing a dynamically added tab.

To evaluate this manually, spin up a narrative from this PR, create a narrative, add a FeatureClusters object, and open the viewer. 

Unfortunately it is not possible to manually evaluate all usage scenarios with the Narrative, as they are not all utilized or demonstrable. To cover those cases I used a small gallery feature which allows embedding a widget in a static html page in order to demonstrate all usage scenarios. That code is not included in this PR. If "storybook" is added to the Narrative, such demonstrations of UI functionality can be included with PRs.

  - see comment below; ui unit tests pass in isolation, but inevitably fail when all run in sequence.

# Dev Checklist:

  - ui unit tests appear unaltered after these changes; the tests are generally non-deterministic - in isolation all tests
    pass, but when all run together, there are inevitably failures.

# Updating Version and Release Notes (if applicable)
